### PR TITLE
docs(button): update Button story and documentation

### LINKED ISSUE #6564: Update the code documentation for Button
This issue is to update the stories that we have for the Button component in our storybook. This work includes the following:

- [x] Update each story in `Button-story.js` to Component Story Format
  - There is an optional codemod that you can use from storybook:
  - `npx -p @storybook/cli@next sb migrate storiesof-to-csf --glob='packages/react/src/Button/Button-story.js'`
- [ ] Create a `Button.stories.mdx` file and fill it out according to the template below
  - Use the `includeStories` option in `Button-story.js` to optionally include/exclude stories
  - You can refer to stories by their story id from `Button-story.js` in `Button.stories.mdx`
- [ ] Document any usage mishaps for v11

<details>
<summary>Template</summary>

```mdx
import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import {
  Button,
  ButtonSkeleton
} from '../Button';

<Meta
  title="Button" 
  component={Button}
  subcomponents={{
    ButtonSkeleton
  }}
/>

> How to build a(n) [component name] using React.

## Overview
<!--
This should include the default use-case(s) for a component, at most 2.
This should be the most common way the component is used.
It should summarize what this common is for and provide snippets for how
to accomplish these common use-cases
-->

## [Use case, for example, Skeleton state or Sorting]
<!--
Each use-case of this component becomes its own section. For example,
if an accordion had a skeleton then you would name this section Skeleton.

If the component had a specific piece of functionality or behavior, like
how you can sort with a Data Table, then Sorting would be the name of this
section.

This section should include an explainer for the use-case and when to use it,
along with a code snippet that someone could bring into their project.
-->

## Component API

### [Component name] [prop name]
<!--
Used to clarify prop usage, provide additional info or snippets
-->


## References
<!--
A place for references, links, or a way clarify points made earlier
in the documentation.
-->

## Feedback

Help us improve these docs by [editing this file on GitHub](https://github.com/carbon-design-system/carbon/edit/master/packages/react/src/components/Accordion/Accordion.stories.mdx)
```
</details>

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/6564

Updates `Button` to new CSF format and adds in prop documentation 

**Done**
- [x] Update to CSF
- [x] Create `Button.mdx` and fill out with template
- [x] Document any discrepancies for v11

### Changelog

**New**
- Added `react/sort-prop-types` as a warning so that our `PropType` definitions are in alphabetical order. THat way the Prop tables will also be in alphabetical order. 
- `Docs` tab now includes detailed descriptions of each `prop` as well as examples and code snippets
- Set default prop values for `tooltipAlignment` and `tooltipPosition` so `icon-only-buttons` require two less props to be specified by default. 

**Changed**

- Sorted `PropType` definitions alphabetically 

#### Testing / Reviewing

Check updated docs for grammar usage and that the examples look correct

#### Questions for reviewers:

- is the `set of buttons` story is still up-to-date?
- Any other examples we want to show? 

#### Issues for V11

- [ ] Button skeleton uses deprecated `small` instead of `size`

### CODE REVIEW

**Comment by tw15egan** [comment]
@joshblack question about knobs: are we removing them in favor of showing all examples in the docs? Ran into some issues trying to get the `IconOnly` button to render due to the knobs. Also, the `Show Code` is not super helpful since it just shows `{...props.iconOnly()}`

Also wondering if we should just focus on the props + usage in the docs, rather than "patterns". In this case, the `Set of Buttons` pattern doesn't seem too helpful and has no props specific to that variation.

**Review by andreancardona — APPROVED** [review]

**Comment by tw15egan** [comment]
@andreancardona I just updated this to match the epic / other docs PR's - mainly just removed some extra stories, and put each variation as it's own story.

**Comment by andreancardona** [comment]
> ther docs PR'

Looks good! @tw15egan does percy need to pass?

**Comment by tw15egan** [comment]
@andreancardona approved them, differences were due to changing the stories to show the variants vs the old patterns

**Comment by andreancardona** [comment]
> @andreancardona approved them, differences were due to changing the stories to show the variants vs the old patterns

sounds good :)

File: config/eslint-config-carbon/plugins/react.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ module.exports = {
     // react-hooks
     'react-hooks/rules-of-hooks': 2,
     'react-hooks/exhaustive-deps': 2,
+
+    // prop-types
+    'react/sort-prop-types': 1,
```
Comment (joshblack) [inline config/eslint-config-carbon/plugins/react.js]:
```suggestion
```

Should be configured as an error now after the prop type refactor!

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by aledavila — APPROVED** [review]

File: config/eslint-config-carbon/plugins/react.js (hunk 1 above)
Comment (tw15egan) [reply config/eslint-config-carbon/plugins/react.js]: Fixed 👍

**Comment by alisonjoseph** [comment]
Looks like some conflicts because the new `ButtonSet` component got merged in

**Comment by tw15egan** [comment]
@alisonjoseph should be good now 👍

**Review by joshblack — APPROVED** [review]

**Pull request merged**